src/prefect/_internal/concurrency/api.py

```diff
@@ -0,0 +1,141 @@
+import abc
+import asyncio
+import concurrent.futures
+from typing import Awaitable, Callable, Optional, TypeVar, Union
+
+from typing_extensions import ParamSpec
+
+from prefect._internal.concurrency.runtime import get_runtime_thread
+from prefect._internal.concurrency.supervisors import (
+    AsyncSupervisor,
+    Call,
+    Supervisor,
+    SyncSupervisor,
+    get_supervisor,
+)
+
+P = ParamSpec("P")
+T = TypeVar("T")
+Future = Union[concurrent.futures.Future, asyncio.Future]
+
+
+def create_call(__fn: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> Call[T]:
+    return Call(fn=__fn, args=args, kwargs=kwargs)
```

Comment:
Is there an advantage of the `create_call` factory versus folks just creating `Call` objects or using `Call.new`?  Maybe a layer of indirection we don't need?

Reply:
Good question. With `Call.new` around it feels less necessary. The best reason I can think of to keep it here is that I think `Call` should be a private implementation detail of the `supervisors` module and this module should contain the full "public API" for managing concurrency.